### PR TITLE
Add clipboard sharing

### PR DESCRIPTION
**Pull request by aledeg**
See #3327

Changes proposed in this pull request:

- Add a clipboard sharing option

How to test the feature manually:

1.
2.
3.

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested
- [ ] unit tests written (optional if too hard)
- [ ] documentation updated

[Additional information can be found in the documentation](https://github.com/FreshRSS/FreshRSS/tree/master/docs/en/developers/04_Pull_requests.md).

When the discussion #3327 started, I thought that the clipboard share
is a good idea. It never occurred to me because I do not use a
smartphone.
It might need some polishing but it seems pretty usable as-is.

### CODE REVIEW

File: app/i18n/nl/gen.php [hunk 1]
```diff
@@ -185,6 +185,7 @@
 	'share' => array(
 		'Known' => 'Known-gebaseerde sites',
 		'blogotext' => 'Blogotext',
+		'clipboard' => 'Clipboard',	// TODO - Translation
```
Comment (Frenzie) [inline app/i18n/nl/gen.php]:
```suggestion
		'clipboard' => 'Klembord',
```

File: p/scripts/main.js [hunk 2]
```diff
@@ -960,6 +960,17 @@ function init_stream(stream) {
...
 
+		el = ev.target.closest('.item.share > a[data-type="clipboard"]');
+		if (el) {	//Clipboard
+			const textarea = document.createElement('textarea');
+			textarea.textContent = el.href;
+			document.body.appendChild(textarea);
+			textarea.select();
+			document.execCommand('copy');
+			document.body.removeChild(textarea);
+			return false;
```
Comment (Frenzie) [inline p/scripts/main.js]: You think https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/writeText is too new?

File: app/i18n/de/gen.php [hunk 3]
```diff
@@ -185,6 +185,7 @@
 	'share' => array(
 		'Known' => 'Known-Seite (https://withknown.com)',
 		'blogotext' => 'Blogotext',
+		'clipboard' => 'Clipboard',	// TODO - Translation
```
Comment (rom-1) [inline app/i18n/de/gen.php]: `'clipboard' => 'Zwischenablage',`

**Review by rom-1 — CHANGES_REQUESTED** [review]

File: p/scripts/main.js (hunk 2 above)
Comment (aledeg) [reply p/scripts/main.js]: I didn't know about it. It seems that it's fine but it is not supported by Internet Explorer.
I was searching if we still support it but couldn't find that information in the doc. If we don't, I'll change that.

Comment (aledeg) [reply p/scripts/main.js]: I found the information and we still support IE.

Comment (Frenzie) [reply p/scripts/main.js]: IE is dead now, cf. what I wrote in https://github.com/FreshRSS/FreshRSS/issues/2159#issuecomment-443409899

**Review by Frenzie — APPROVED** [review]
But that IE thing isn't important. :+1:

**Comment by aledeg** [comment]
I'll see what I can do with the clipboard API. I guess it would be better than what I have done.

**Comment by aledeg** [comment]
@Frenzie Thank you! I didn't know about that API.
It works really well and it's really easy to use it.
I know another project where I will use it.

File: p/scripts/main.js [hunk 4]
```diff
@@ -960,6 +960,12 @@ function init_stream(stream) {
 			return false;
 		}
 
+		el = ev.target.closest('.item.share > a[data-type="clipboard"]');
+		if (el) {	//Clipboard
+			navigator.clipboard.writeText(el.href);
```
Comment (Alkarex) [inline p/scripts/main.js:965]: Could you please test `if (navigator.clipboard) {}` to fail gracefully with e.g. IE11?

Comment (aledeg) [reply p/scripts/main.js:965]: done

File: p/scripts/main.js (hunk 2 above)
Comment (Alkarex) [reply p/scripts/main.js]: Yes, I think it is OK not to have IE11 support for new features. It is better if it does not break/crash though, for now, but that can be tested with feature detection

**Review by Alkarex — APPROVED** [review]

**Comment by Alkarex** [comment]
Nice, works fine :-)
As a potential future improvement, maybe auto-closing the menu and/or provide some kind of feedback could be nice.

**Pull request merged**